index.js

```diff
@@ -0,0 +1,225 @@
+	function hasClass(el, className) {
+	  if (el.classList) {
+  		return el.classList.contains(className);
+	  }	 else {
+	  	return !!el.className.match(new RegExp('(\\s|^)' + className + '(\\s|$)'));
+	  }	    
+	}
+
+	function addClass(el, className) {
+	  if (el.classList) {
+  		el.classList.add(className);
+	  }	    
+	  else if (!hasClass(el, className)) {
+	  	el.className += " " + className;
+	  }
+	}
+
+	function removeClass(el, className) {
+	  if (el.classList) {
+	  	el.classList.remove(className);
+	  }	else if (hasClass(el, className)) {
+	    var reg = new RegExp('(\\s|^)' + className + '(\\s|$)');
+	    el.className=el.className.replace(reg, ' ');
+	  }
+	}
+
+	//board class
+	var Board = function(x,y, drawingBoard) {
+		this.checkErrors(x,y,drawingBoard);
+		this.drawingBoard = drawingBoard;		
+		this.x = x;
+		this.y = y;
+		this.activeItems = [];
+		this.items = this.randomizeBoardPieces(x,y);
+		drawingBoard.draw(this); 
+	};
+
+	Board.prototype.checkErrors = function(x,y,drawingBoard) {
+		if (x <= 0) {
+			throw new Error('Rows must be greater than 0');
+		
+		}
+		if (y <= 0) {
+			throw new Error('Columsn must be greater than 0');
+		}
+		if (x*y % 2 !== 0) {
+			throw new Error('You really should have an even number of pieces, or you will never win');	
+		}
+		if (x*y > 26) {
+			throw new Error('Currently you can not go higher than alphabet letters. 26 Boxes is Max');	
+		}		
+	};
+
+	Board.prototype.selectNode = function(index) {		
+		this.activeItems.push(index);
+		this.compare();
+	};
+
+	Board.prototype.compare = function() {		
+		var items = this.items, activeItems = this.activeItems;
+		if (activeItems.length === 2 && items[activeItems[0]] === items[activeItems[1]]) {
+			this.matchSuccess();
+		} else if (activeItems.length === 2) {			
+			this.reset(true);
+		}
```

Comment:
Since you're using the same condition twice, I would rewrite as

```javascript
if (activeItems.length === 2) {
	if (items[activeItems[0]] === items[activeItems[1]]) {
		this.matchSuccess();
	} else {			
		this.reset(true);
	}
}
```